Add functionality for a banner to promote different event where Kyma is present (#96)
kyma-project/website: docs/banner-modification.md

# Banner modification

> To edit banner on the website you have to make changes in [banners.json](../src/banner/banners.json). This file has following structure:

Reviewer: ```suggestion
To edit the banner on the website you have to make changes in [banners.json](../src/banner/banners.json). This file has the following structure:
```

```json
{
  "bannerDuration": "{BANNER_DURATION}",
  "slides": [
    {
      "text": "{TEXT}",
      "url": "{URL}",
      "openInNewTab": "{OPEN_IN_NEW_TAB}",
      "startDate": "{START_DATE}",
      "endDate": "{END_DATE}",
      "icon": "{ICON_FILE}"
    }
  ]
}
```

Replace these parameters with real values:

> - **{BANNER_DURATION}** is time duration in miliseconds after which banner slide changes to next one if there is more than one slide.

Reviewer: ```suggestion
- **{BANNER_DURATION}** is the time (in milliseconds) after which a banner slide changes to the next one if there is more than one slide.
```

> - **{TEXT}** is the banner's text. One sentence of content is recommended.

Reviewer: ```suggestion
- **{TEXT}** is the banner's text. The content you provide should be no longer than one sentence.
```

- **{URL}** is an optional address. If present, the `Read more` link appears to the right of the banner's text.

> - **{OPEN_IN_NEW_TAB}** indicates whether an optional link is pointing to an internal or external resource. Set it to `false` for external links like `https://www.google.com/`, and to `true` for internal links like `/blogs`. Remove this line if a link is not present. If this parameter is not provided it defaults to external link.

Reviewer: ```suggestion
- **{OPEN_IN_NEW_TAB}** indicates whether an optional link is pointing to an internal or external resource. Set it to `false` for external links like `https://www.google.com/`, and to `true` for internal links like `/blogs`. Remove this line if a link is not present. If you do not provide this parameter, it points to the external link by default.
```

- **{START_DATE}** is the date from which a particular banner slide starts to appear on the website, counting from midnight. Use the `DD-MM-YYYY` format.
- **{END_DATE}** is the date up to which a particular banner slide appears, counting to 23:59. Use the `DD-MM-YYYY` format.

> - **{ICON_FILE}** is a path of an icon used in the banner slide. Store icons in the [`assets`](../src/banner/assets) folder. It is an optional parameter.

Reviewer: ```suggestion
- **{ICON_FILE}** is a path to an icon used in the banner slide. Store icons in the [`assets`](../src/banner/assets) folder. It is an optional parameter.
```

    "startDate": { START_DATE },
    "endDate": { END_DATE },
    "icon": { ICON_FILE }
  }
]
```

Replace these parameters with real values:

- `{TEXT}` is the banner's text.
- `{URL}` is optional address - if present, `READ MORE` link appears to the right of the banner's text.
- `{INTERNAL}` indicates whether optional link is pointing to some internal or external resource. Set it to `false` for external link like `https://www.google.com/`, and to `true` for internal links like `/blogs`. Remove this line if link is not present.
- `{START_DATE}` is the starting day from which particular banner slide will be shown, counting from 00:00 at night. Use `DD-MM-YYYY` format.
- `{END_DATE}` is the ending day up to which particular banner slide will be shown, counting to 23:59 at night. Use `DD-MM-YYYY` format.
- `{ICON}` is filename of icon used in the banner slide. Icons are stored in [./src/banner/assets](../src/banner/assets) folder. It is an optional parameter.

`{TEXT}`, `{START_DATE}`, and `{END_DATE}` are mandatory parameters. `{INTERNAL}` is also mandatory if `{URL}` is provided.

To show banner for one full day make sure `{START_DATE}` and `{END_DATE}` are equal.

> ## Icon

Reviewer: ```suggestion
## Add an icon
```

To use it put desired image file in [./src/banner/assets](../src/banner/assets) folder. This image will be scaled to fit into 30px wide, 45px high box, while preserving its original ascpect ratio.

Image must be in `jpeg`, `png`, or `gif` format.

> ## Multiple slides

Reviewer: ```suggestion
## Add multiple slides
```

> To show multiple slides that are perodically changing like in typical carousel add more sections enclosed by `{` and `}`, as in this example:

Reviewer: ```suggestion
To show multiple slides that change periodically, like in a typical carousel, add more sections enclosed in curly brackets and separate them with a comma, as in this example:
```

```json
[
  {
    "text": "A flexible and easy way to connect and extend enterprise",
    "startDate": "01-11-2018",
    "endDate": "30-11-2019"
  },
  {
    "text": "Kyma is an open-source project designed natively on Kubernetes.",
    "url": "https://google.com",
    "external": true,
    "startDate": "01-10-2018",

>     "endDate": " 30-11-2018",

Reviewer: ```suggestion
    "endDate": "30-11-2018",
```

    "icon": "icons_filename.jpeg"
  },
  {
    "text": "Some kind of internal link",
    "url": "/blog",
    "external": false,
    "startDate": "05-11-2018",

>     "endDate": " 05-11-2018",

Reviewer: ```suggestion
    "endDate": "05-11-2018",
```

    "icon": "example3.png"
  }
]
```

> ## No banner

Reviewer: ```suggestion
## Remove the banner
```

> If no banner should be shown change [banners.json](../src/banner/banners.json) into

Reviewer: ```suggestion
If you want to remove the banner from the website, change [banners.json](../src/banner/banners.json) as follows:
```
